decentraland/screens/EthereumSignUpScreen.js

```diff
@@ -21,12 +22,16 @@ export class EthereumSignUpScreen extends React.Component {
       // See more: https://github.com/tasitlabs/tasit/issues/42
       const account = Account.create();
       setAccount(account);
+      showInfo(`Account generated`);
 
       const { address: accountAddress } = account;
 
-      await fundAccount(accountAddress);
-      await approveManaSpending(account);
-      showInfo(`Account created and funded!`);
+      await fundAccountWithEthers(accountAddress);
+      showInfo(`Account funded with ethers`);
+      const fundWithMana = fundAccountWithMana(accountAddress);
+      const approveMarketplace = approveManaSpending(account);
+      await Promise.all([fundWithMana, approveMarketplace]);
```

Comment:
Is it an issue if an approve transaction happens before the account has MANA?

Reply:
> I've made a tiny performance improvement running `marketplace approval` and `transfer mana` transactions in parallel.

This question is related to your note

Reply:
No, not really. The `approve` function doesn't check the `msg.sender` balance.

Reply:
Hmm, that's weird. Good to know. Okay, resolving this and we can keep that in mind.

Reply:
I've looked `MANAToken`  and `OpenZeppelin` ERC20/ERC721 codes and none of them require balance.

Reply:
Thanks. Yep, I checked OpenZeppelin too. K, good to know!